src/lib/dns.coffee

```diff
@@ -114,42 +112,27 @@ module.exports = (dnschain) ->
             # At some point we may still want to support this though.
             q = req.question[qIdx=0]
             q.name = q.name.toLowerCase()
-            
+
             ttl = Math.floor(Math.random() * 3600) + 30 # TODO: pick an appropriate TTL value!
             @log.debug "received question", q
 
-            if /\.(bit|p2p)$/.test q.name
-                if S(q.name).endsWith '.bit'
-                    nmcDomain = @namecoinizeDomain q.name
-                    resolver = 'nmc'
-                else
-                    # TODO: bdns-izeDomain
-                    nmcDomain = S(q.name).chompRight('.p2p').s
-                    resolver = 'bdns'
-                
-                @log.debug gLineInfo("resolving via #{resolver}..."), {nmcDomain:nmcDomain, q:q}
-
-                @dnschain[resolver].resolve nmcDomain, (err, result) =>
-                    # @log.warn gLineInfo('blah!'), {result: result}
+            chainTLDs = new RegExp('\\.('+_.map(@dnschain.chains, 'tld').join('|')+')$')
+            if chainTLDs.test q.name
+                tld = q.name.split('.').pop()
+                resolver = _.find @dnschain.chains, (c) ->
+                   c.tld? and c.tld == tld
```

Comment:
If you coffeelint this file using grunt (and the `coffeelint.json` in the main directory, as you should be doing) you'll notice that this is an error (inconsistent indentation, 3 spaces instead of 4).